mongodb: change openssl option from optional to recommended

briananthonymccarthy opened the pull request:
Changed the openssl option from 'optional' to 'recommended' so that mongodb installs with an ssl-enabled binary by default.  This will require bottle rebuilds so that ssl is enabled in the default binaries. Based on our interpretation of the guidelines, we assume this will happen along with this change.

apjanke commented:
You will need to increment the formula's `revision` to get rebuilt bottles and trigger upgrades of existing installations. `brew` isn't smart enough to determine which in-formula changes require rebuilding & reinstallation on its own.

Just stick a `revision 1` line inside the `stable do` block, probably right after the `sha256` line, and run `brew audit --strict mongodb` to make sure it likes the arrangement.

DomT4 commented:
> Based on our interpretation of the guidelines, we assume this will happen along with this change.

Who is "we" in this case? The `mongodb` organisation?

briananthonymccarthy commented:
Thank you.  Yes, I work for mongodb.  I will try incrementing the revision as you suggested and update this request.

apjanke commented:
Sorry if I wasn't clear. The `revision` to increment should go in the main formula, inside the `stable do` block, not inside the `bottle` block. The stuff in the `bottle` block is all handled automatically by BrewTestBot, and shouldn't be edited by humans.

It should look like this:

```


  stable do
    url "https://fastdl.mongodb.org/src/mongodb-src-r3.2.4.tar.gz"
    sha256 "b60743cc641de975c38e6e69ebbef60059ee9fe176cdd98bfab8d5c844dab42c"
    revision 1

    go_resource "github.com/mongodb/mongo-tools" do
      url "https://github.com/mongodb/mongo-tools.git",
        :tag => "r3.2.4",
        :revision => "eacbffc3c185686fb572c9efa8fcfaf9e9fb5c32"
    end
```

We should move on https://github.com/Homebrew/brew/issues/38; this is a common misunderstanding and is costing both maintainer and contributor time. Sorry for the inconvenience.

briananthonymccarthy commented:
Thank you.  I actually did try that first, but the 'brew audit' failed with the below output:

`MacBook-Pro-6:homebrew-core brianmccarthy$ brew audit --strict mongodb
Error: undefined method `revision' for #<SoftwareSpec:0x007f873322d8e8>
Please report this bug:
    https://git.io/brew-troubleshooting`

DomT4 commented:
Oh, you can't put `revision` lines inside blocks. We explicitly don't support stable/devel/head-explicit revisions at this point in time. The `revision 1` line should be underneath the `homepage "https://www.mongodb.org/"` line.

apjanke commented:
Aha! Sorry for misleading you. This DSL is a bit complicated. Thanks Dom.

So, like this.

```

class Mongodb < Formula
  desc "High-performance, schema-free, document-oriented database"
  homepage "https://www.mongodb.org/"
  revision 1

  stable do
    url "https://fastdl.mongodb.org/src/mongodb-src-r3.2.4.tar.gz"
    sha256 "b60743cc641de975c38e6e69ebbef60059ee9fe176cdd98bfab8d5c844dab42c"
```

BTW, since there's only a stable release here, and no devel or HEAD release, you can remove the `stable do` block entirely and just put its contents in the main formula like this.

```
class Mongodb < Formula
  desc "High-performance, schema-free, document-oriented database"
  homepage "https://www.mongodb.org/"
  url "https://fastdl.mongodb.org/src/mongodb-src-r3.2.4.tar.gz"
  sha256 "b60743cc641de975c38e6e69ebbef60059ee9fe176cdd98bfab8d5c844dab42c"
  revision 1

  go_resource "github.com/mongodb/mongo-tools" do
    url "https://github.com/mongodb/mongo-tools.git",
      :tag => "r3.2.4",
      :revision => "eacbffc3c185686fb572c9efa8fcfaf9e9fb5c32"
  end
```

Also, around line 53, `brew audit` doesn't like this string interpolation as the only thing inside quotes.
```
        ENV["LIBRARY_PATH"] = "#{Formula["openssl"].opt_lib}"
        ENV["CPATH"] = "#{Formula["openssl"].opt_include}"
```
```
$ brew audit --strict mongodb
==> brew style mongodb
== mongodb.rb ==
C: 53: 31: Prefer to_s over string interpolation.
C: 54: 24: Prefer to_s over string interpolation.

1 file inspected, 2 offenses detected
```

Do you mind fixing that while you're in here? Should be this instead, I think.
```
        ENV["LIBRARY_PATH"] = Formula["openssl"].opt_lib
        ENV["CPATH"] = Formula["openssl"].opt_include
```

DomT4 commented:
> This DSL is a bit complicated. Thanks Dom.

No worries at all. I've forgotten and fallen into that pit a couple of times myself.

briananthonymccarthy commented:
Thanks everyone!  I pushed with 'revision 1' in the right place before I caught that last comment on the string interpolation, but I'll fix that and push again.

briananthonymccarthy commented:
OK, just incorporated all of the above suggestions.  Removed the stable/do block and fixed the string interpolation.  brew audit now passes with no offenses:

```
brew audit --strict mongodb
==> brew style mongodb

1 file inspected, no offenses detected
```

Thank you everyone for all the help and suggestions.

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!
